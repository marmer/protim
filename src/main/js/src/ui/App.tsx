import * as React from 'react';
import {TimeManagementView} from './TimeManagementView';

class App extends React.Component {
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src="./logo.svg" className="App-logo" alt="logo"/>
                    <h1 className="App-title">Welcome to Protim</h1>
                </header>
                <TimeManagementView/>
            </div>
        );
    }
}

export default App;
