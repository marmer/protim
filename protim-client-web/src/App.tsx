import * as React from 'react';
import {TimeManagementView} from './ui/TimeManagementView';
import {UserView} from "./ui/UserView";

class App extends React.Component {
    render() {
        return (
            <div className="App">
                <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
                    <a className="navbar-brand" href="#">Protim</a>
                </nav>
                <UserView username="admin"/>
                <TimeManagementView/>
            </div>
        );
    }
}

export default App;
