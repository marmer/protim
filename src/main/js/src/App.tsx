import * as React from 'react';
import './App.css';
import {BookingDayView} from './BookingDayView';

class App extends React.Component {
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <img src="./logo.svg" className="App-logo" alt="logo"/>
                    <h1 className="App-title">Welcome to React</h1>
                </header>
                <p className="App-intro">
                    To get started, edit <code>src/App.tsx</code> and save to reload.
                    <div className="alert alert-primary" role="alert">
                        This should be pretty green if bootstrap loads. Otherwhise something went wrong ^^
                    </div>
                </p>
                <BookingDayView/>
            </div>
        );
    }
}

export default App;
