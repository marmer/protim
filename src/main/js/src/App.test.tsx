/// <reference path="../node_modules/jest-enzyme/lib/index.d.ts" />

import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './App';
import {shallow, ShallowWrapper} from 'enzyme';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
});

describe(`<${App.name} />`, () => {
    let tree: ShallowWrapper;
    beforeAll(() => {
        tree = shallow(<App/>);
    });

    it('View with current day is displayed', () => {
        expect(tree.find('BookingDayView')).toExist();
    });

    it('An important day was set for mankind', () => {
        expect(tree.find('BookingDayView').prop('date')).toEqual('1979-10-12');
    });

    it('snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
});