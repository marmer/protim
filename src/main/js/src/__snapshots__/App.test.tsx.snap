// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="./logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.tsx
    </code>
     and save to reload.
    <div
      className="alert alert-primary"
      role="alert"
    >
      This should be pretty colored if bootstrap loads. Otherwhise something went wrong ^^
    </div>
  </p>
  <BookingDayView
    date="1979-10-12"
  />
</div>
`;
